pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,2311 @@
+                Self::update_validators_pool(validator.clone(), validator_state.total);
+            }
+
+            <Total<T>>::mutate(|x| *x += amount);
+            <ValidatorState<T>>::insert(&validator, validator_state);
+            <NominatorState<T>>::insert(&nominator_acc, nominator_state);
+
+            Self::deposit_event(Event::Nomination(
+                nominator_acc,
+                amount,
+                validator,
+                validator_new_total,
+            ));
+
+            Ok(().into())
+        }
+        /// Revoke an existing nomination
+        #[pallet::weight(T::WeightInfo::nominator_denominate())]
+        pub fn nominator_denominate(
+            origin: OriginFor<T>,
+            validator: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            Self::nominator_revokes_validator(ensure_signed(origin)?, validator, false)
+        }
+        /// Quit the set of nominators and, by implication, revoke all ongoing nominations
+        #[pallet::weight(T::WeightInfo::nominator_denominate_all())]
+        pub fn nominator_denominate_all(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let acc = ensure_signed(origin)?;
+
+            let nominator = <NominatorState<T>>::get(&acc).ok_or(Error::<T>::NominatorDNE)?;
+
+            for bond in nominator.nominations.0 {
+                Self::nominator_revokes_validator(acc.clone(), bond.owner.clone(), true)?;
+            }
+
+            Ok(().into())
+        }
+        /// Bond more for nominators with respect to a specific validator
+        #[pallet::weight(T::WeightInfo::nominator_bond_more())]
+        pub fn nominator_bond_more(
+            origin: OriginFor<T>,
+            validator: T::AccountId,
+            more: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let nominator = ensure_signed(origin)?;
+            let mut nominations =
+                <NominatorState<T>>::get(&nominator).ok_or(Error::<T>::NominatorDNE)?;
+            let mut validator_state =
+                <ValidatorState<T>>::get(&validator).ok_or(Error::<T>::ValidatorDNE)?;
+            let _ = nominations
+                .inc_nomination(validator.clone(), more)
+                .ok_or(Error::<T>::NominationDNE)?;
+
+            let nominator_free_balance = T::Currency::free_balance(&nominator);
+            ensure!(
+                nominator_free_balance >= nominations.total,
+                Error::<T>::InsufficientBalance
+            );
+
+            // T::Currency::reserve(&nominator, more)?;
+            T::Currency::set_lock(
+                STAKING_ID,
+                &nominator,
+                nominations.total,
+                WithdrawReasons::all(),
+            );
+
+            let before = validator_state.total;
+            validator_state.inc_nominator(nominator.clone(), more);
+            let after = validator_state.total;
+            <Total<T>>::mutate(|x| *x += more);
```

Comment:
use safe math or saturating operators :)

Reply:
:+1:  Moved to saturating_add()